include/GafferSceneUI/Private/Inspector.h

```diff
@@ -0,0 +1,238 @@
+///   works in. We think this information can be stored in a dedicated TransformHistory
+///   class provided by SceneAlgo, avoiding any need to specialise Inspector::Result.
+class GAFFERSCENEUI_API Inspector : public IECore::RefCounted, public boost::signals::trackable
+{
+
+	public :
+
+		IE_CORE_DECLAREMEMBERPTR( Inspector );
+		IE_CORE_FORWARDDECLARE( Result );
+
+		/// The name of the property being inspected.
+		const std::string &name() const;
+
+		/// Called by the UI to inspect the property in the current context.
+		ResultPtr inspect() const;
+
+		using InspectorSignal = boost::signal<void ( Inspector * )>;
+		/// Emitted when the property queried by the inspector has changed.
+		/// The UI should use this to schedule a refresh.
+		InspectorSignal &dirtiedSignal();
+
+	protected :
+
+		Inspector( const std::string &name, const Gaffer::PlugPtr &editScope );
+
+		/// Methods to be implemented in derived classes
+		/// ============================================
+		///
+		/// The `inspect()` method delegates to several virtual methods that
+		/// should be implemented by derived classes. Inspection starts by
+		/// generating a history for the computation of the property, and then
+		/// traverses the history making additional queries at various points.
+
+		/// Must be implemented to return the history for the property being
+		/// inspected. Should return null if the property does not exist.
+		virtual GafferScene::SceneAlgo::History::ConstPtr history() const = 0;
+
+		/// Must be implemented to return the value of the property at this
+		/// point in the history. Called with `history->context` as the current
+		/// context.
+		/// \todo Perhaps this should be available directly from the history
+		/// base class?
+		virtual IECore::ConstObjectPtr value( const GafferScene::SceneAlgo::History *history ) const = 0;
+
+		/// Should be implemented by derived classes to return the source for
+		/// the value authored at this point in the history. Optionally,
+		/// `editWarning` may be assigned a warning that will be shown to the
+		/// user when editing this plug. Called with `history->context` as the
+		/// current context. Default implementation returns null.
+		virtual Gaffer::ValuePlugPtr source( const GafferScene::SceneAlgo::History *history, std::string &editWarning ) const;
+
+		using EditFunction = std::function<Gaffer::ValuePlugPtr ()>;
+		using EditFunctionOrFailure = boost::variant<EditFunction, std::string>;
+		/// Should be implemented to return a function that will acquire
+		/// an edit from the EditScope at the specified point in the history.
+		/// If this is not possible, should return an error explaining why
+		/// (this is typically due to `readOnly` metadata). Called with `history->context`
+		/// as the current context.
+		///
+		/// > Note : Where an EditScope already contains an edit, it is expected
+		/// > that this will be dealt with in `source()`, returning a result
+		/// > that edits the processor itself.
```

Comment:
It's because acquiring an edit means modifying the graph in the majority of cases, and we only want to do that when the user explicitly chooses to do so.